com.unity.shadergraph/Documentation~/Refract-Node.md

```diff
@@ -0,0 +1,42 @@
+# Refract Node
+
+## Description
+
+Generates a refraction [Ibn Sahl-Snell-Descartes's Law](https://en.wikipedia.org/wiki/Snell%27s_law). Based on an **Incident** vector, the **Normal** of the surface and **Eta** we produce the refracted vector.
+
+Based on Ibn Sahl's Law, for a given refractive index of the medium, we can reach an angle where all the surface behave like a perfect mirror, that's why in **Safe** Mode generate a null vector, this implementation is here to not double the reflection intensity which is handle by Fresnel.
+
+## Ports
+
+| Name        | Direction           | Type  | Binding | Description |
+|:------------ |:-------------|:-----|:---|:---|
+| Incident      | Input | Vector | None | Incident normalied vector to the surface (From surface to source, for instance from pixel to eye). |
+| Normal      | Input | Vector | None | Normal normalized of the surface |
+| Eta      | Input | Float    | None | The ratio of refractive index (Eta == (source refractive index)/(medium refractive index)), For instance for an interaction from air to glass **Eta** will be 1.0/1.5 and for an interaction from glass to air **Eta** will be 1.5/1.0. |
+| Out | Output      |  Vector | None | Refracted vector |
+
+## Controls
+
+| Name        | Type           | Options  | Description |
+|:------------ |:-------------|:-----|:---|
+| Mode      | Dropdown | CriticalAngle, Safe | Select if the implementation will protect against critical angle refraction which will produce a NaN (Square root of a negative number), if the Safe mode is used when we reach the critical angle a null vector is returned. |
```

Comment:
I don't understand what these two modes are supposed to do. Does selecting "CriticalAngle" allow for returning the NaN result, and "Safe" prevents this?

Reply:
Yes that's it, the Safe one have a small extra cost.
For example if you always do Air-Medium interaction you'll never have NaN, sor "cheaper" "CritialAngle" is the way to go.
But for Medium-Air it's either the user check the value to switch or always use the "Safe".

Reply:
> Select **Safe** to prevent the Refract node from returning a NaN result when there is a risk of critical angle refraction. **Safe** will return a null vector result, instead. Select **CriticalAngle** to avoid this check for a potential NaN result. 

Please change the text to the above.